src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -249,16 +248,6 @@ class BashBinary(BinaryPath):
     DEFAULT_SEARCH_PATH = SearchPath(("/usr/bin", "/bin", "/usr/local/bin"))
 
 
-class PythonBinary(BinaryPath):
```

Comment:
This type was meant to be a public API (similar to `BashBinary`) for `@rule` authors to use to write scripts.

Including a deprecation warning of some sort would be good at the very least. Better yet might be to ducktype the old type to support being implemented in terms of the new type (or giving the new type the old type's name).

Reply:
They aren't necessarily 1:1 compatible, since the new type requires immutable input digests (sometimes, so I fear it's pass locally but fail down the road)

Reply:
Ok... then a deprecation message perhaps, pointing to the new type to use instead?

Reply:
Yeah let me noodle this one.

Reply:
By deprecation message, do you mean using something from `deprecated.py`? Or a message in the changelog?

Reply:
Both.